From 253f7ff0a31708f9bd037605d907c552d5a9aad5 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 10 May 2017 12:12:35 +0300
Subject: [PATCH 0760/1200] soc: apn806: add soc early init to u-boot 2018

SoC early init is required in order to print the
SoC information at the beginning of u-boot bootflow.

Note that the added part is only for the usage of
sample at reset register init.

Change-Id: I01c82075f7eb7afaa3bae9a2f49b93382e4548b6
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39280
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57178
---
 arch/arm/mach-mvebu/armada8k/soc.c    | 11 +++++++++++
 board/Marvell/mvebu_armada-8k/board.c | 10 +++++++++-
 2 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-mvebu/armada8k/soc.c b/arch/arm/mach-mvebu/armada8k/soc.c
index 586e844043..933eb2ba49 100644
--- a/arch/arm/mach-mvebu/armada8k/soc.c
+++ b/arch/arm/mach-mvebu/armada8k/soc.c
@@ -8,6 +8,7 @@
 #include <common.h>
 #include <asm/io.h>
 #include <asm/arch/soc.h>
+#include <mvebu/mvebu_chip_sar.h>
 
 #define CP_DEV_ID_STATUS_REG		(MVEBU_REGISTER(0x2400240))
 #define DEVICE_ID_STATUS_MASK		0xffff
@@ -142,6 +143,16 @@ void soc_print_device_info(void)
 		printf("CP%x-A%d\n", cp_type, cp_rev);
 }
 
+int soc_early_init_f(void)
+{
+#ifdef CONFIG_MVEBU_SAR
+	/* Sample at reset register init */
+	mvebu_sar_init();
+#endif
+
+	return 0;
+}
+
 #ifdef CONFIG_ARCH_MISC_INIT
 int arch_misc_init(void)
 {
diff --git a/board/Marvell/mvebu_armada-8k/board.c b/board/Marvell/mvebu_armada-8k/board.c
index ab8a86ad91..6aef04b215 100644
--- a/board/Marvell/mvebu_armada-8k/board.c
+++ b/board/Marvell/mvebu_armada-8k/board.c
@@ -17,9 +17,17 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+int __soc_early_init_f(void)
+{
+	return 0;
+}
+
+int soc_early_init_f(void)
+			__attribute__((weak, alias("__soc_early_init_f")));
+
 int board_early_init_f(void)
 {
-	/* Nothing to do (yet), perhaps later some pin-muxing etc */
+	soc_early_init_f();
 
 	return 0;
 }
-- 
2.22.0

